pkg/apis/pipeline/v1/container_validation.go

```diff
@@ -91,6 +91,9 @@ func (ref *Ref) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 // RefNameLikeUrl checks if the name is url parsable and returns an error if it isn't.
 func RefNameLikeUrl(name string) error {
-	_, err := url.ParseRequestURI(name)
-	return err
+	schemeRegex := regexp.MustCompile(`[\w-]+:\/\/*`)
```

Comment:
The regex fixes this and limits the constraint to `<a-scheme>://...` as required in the TEP. The detailed parsing is conducted by the resolver.